pkg/cloudencrypt/dual.go

```diff
@@ -0,0 +1,96 @@
+package cloudencrypt
+
+import (
+	"context"
+)
+
+const (
+	mapKeySecretKey  = "secretKey"
+	mapKeyCipherText = "ciphertext"
+)
+
+// DualEncryptor implements Encryptor by encrypting the value using the NativeEncryptor
+// and then encrypting the secret key using the provided encryptor.
+type DualEncryptor struct {
+	encryptor Encryptor
+}
+
+func NewDualEncryptor(ctx context.Context, encryptor Encryptor) (*DualEncryptor, error) {
+	return &DualEncryptor{
+		encryptor: encryptor,
+	}, nil
+}
+
+func (encryptor *DualEncryptor) Encrypt(ctx context.Context, plaintext []byte, metadata ...MetadataKV) ([]byte, error) {
+	// Generate a random secret key
+	secretKey, err := generateSecretKey()
+	if err != nil {
+		return nil, err
+	}
+
+	nativeEncryptor, err := NewNativeEncryptor(secretKey)
+	if err != nil {
+		return nil, err
+	}
+
+	defer nativeEncryptor.Close()
+
+	// Encrypt given plaintext using the random secret key
+	ciphertext, err := nativeEncryptor.Encrypt(ctx, plaintext, metadata...)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Put this into function or
```suggestion
	// Encrypt given plaintext using the random secret key
	ciphertext, err := nativeEncryptor.Encrypt(ctx, plaintext, metadata...)
	if err != nil {
         	nativeEncryptor.Close()
		return nil, err
	}
	nativeEncryptor.Close()
```
There is no need to have it open when using nested encryptor and the job of nativeencryptor is finished